activerecord/test/models/company.rb

```diff
@@ -26,6 +26,9 @@ def arbitrary_method
   def private_method
     "I am Jack's innermost fears and aspirations"
   end
+
+  class SpecialCo < Company
+  end
```

Comment:
Do we need a new model or can we use `Namespaced::Company` to reproduce the error as well?

```ruby
Namespaced::Company.new(type: "Firm")
```